asmcomp/x86_proc.ml

```diff
@@ -16,20 +16,9 @@
 open X86_ast
 
 type system =
-  (* 32 bits and 64 bits *)
   | S_macosx
   | S_gnu
   | S_cygwin
-
-  (* 32 bits only *)
-  | S_solaris
-  | S_win32
-  | S_linux_elf
-  | S_bsd_elf
-  | S_beos
```

Comment:
The native backend isn't enabled, but that's down to a lack of testing rather than a lack of support - Haiku (BeOS) is also 32/64 bit.